packages/cli/src/index.js

```diff
@@ -83,14 +86,46 @@ const loadDotEnvDefaultsMiddleware = () => {
   })
 }
 
+const updateCheckerMiddleware = (argv) => {
+  const excludedCommands = ['update', 'upgrade', 'ts-to-js']
+  if (excludedCommands.includes(argv._[0])) {
+    return
+  }
+
+  if (update.shouldShow()) {
+    process.on('exit', () => {
+      update.showUpgradeMessage()
+    })
+  }
+
+  if (update.shouldCheck()) {
+    // TODO: Decide how best to call this now it's no longer a command
+    // const logFile = fs.openSync(
+    //   path.join(getPaths().generated.base, 'update-checker.log'),
+    //   'a'
+    // )
+    // const child = spawn('yarn', ['./lib/updateRunner.js'], {
+    //   detached: true,
+    //   stdio: ['ignore', logFile, logFile],
+    //   cwd: getPaths().base,
+    //   shell: process.platform === 'win32',
+    // })
+    // child.unref()
```

Comment:
This requires making a file called `updateCheck.js` (next to `update.js`) with the following contents:


```js
import { check } from './update'

check()
```

And I don't mind adding the stderr back:

```suggestion
    const stdout = fs.openSync(
      path.join(getPaths().generated.base, 'updateCheckStdout.log'),
      'a'
    )
    
    const stderr = fs.openSync(
      path.join(getPaths().generated.base, 'updateCheckStderr.log'),
      'a'
    )


    const child = spawn(
      'yarn',
      ['node', path.join(__dirname, 'lib', 'updateCheck.js')],
      {
        detached: true,
        stdio: ['ignore', stdout, stderr],
      }
    )

    child.unref()
```

Reply:
Thanks for this it was mainly the `['node', path.join(__dirname, 'lib', 'updateCheck.js')]` that tripped me up. I've implemented as you suggested but I've kept in the the shell conditionally on windows. We'll need to ensure the CI passes without it because it would not before.